pkg/model/lattice/targetgroup.go

```diff
@@ -29,13 +30,15 @@ type TargetGroupSpec struct {
 }
 
 type TargetGroupConfig struct {
-	Port            int32  `json:"port"`
-	Protocol        string `json:"protocol"`
-	ProtocolVersion string `json:"protocolversion"`
-	VpcID           string `json:"vpcid"`
-	IpAddressType   string `json:"ipaddresstype"`
-	EKSClusterName  string `json:"eksclustername"`
-	IsServiceImport bool   `json:"serviceimport"`
+	Port              int32                         `json:"port"`
+	Protocol          string                        `json:"protocol"`
+	ProtocolVersion   string                        `json:"protocolversion"`
+	VpcID             string                        `json:"vpcid"`
+	IpAddressType     string                        `json:"ipaddresstype"`
+	EKSClusterName    string                        `json:"eksclustername"`
+	IsServiceImport   bool                          `json:"serviceimport"`
+	HealthCheckConfig *vpclattice.HealthCheckConfig `json:"healthCheckConfig"`
```

Comment:
It looks convenient to reuse generated aws-sdk health check config, but it does not align with models package. I'm not sure what the approach should be, my current understanding:
k8s spec -> lattice models -> aws-sdk models -> lattice calls

So lattice models dont really use aws-sdk, we do transform to aws-sdk models in the deploy package, in managers. It might cause code bloat in terms of simple 1:1 models mapping, but we can do something smart about it.

Reply:
I intend the structure of health check CRD spec to be exactly the same as lattice health check config, to remove any confusion and exactly follow lattice behavior on default values, etc. With this in mind, I still think it is best mapped to the lattice health check config itself. Since lattice models define how we want resources to look like in lattice, I think it kinda aligns.